### PR TITLE
Misc usability issues.

### PR DESCRIPTION
**Pull request by gsaluja9**

### CODE REVIEW

File: aperturedb/Utils.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def __init__(self, connector: Connector, verbose=False):
         self.verbose = verbose
 
     def __repr__(self):
-        return self.status()
+        return f"{id(self)=}"
```
Comment (drewaogle) [inline aperturedb/Utils.py:123]: We will need to bump min python to 3.8 to use the {var=}

File: aperturedb/Connector.py [hunk 2]
```diff
@@ -167,11 +167,7 @@ def authenticate(self, shared_data, user, password, token):
         if not self.authenticated:
             if shared_data.session is None:
                 self.shared_data.lock = Lock()
-                try:
-                    self._authenticate(user, password, token)
-                except Exception as e:
-                    raise UnauthenticatedException(
-                        "Authentication failed:", str(e))
+                self._authenticate(user, password, token)
```
Comment (drewaogle) [inline aperturedb/Connector.py:170]: This means that authenticate will just return on a failed authenticate?

Comment (gsaluja9) [reply aperturedb/Connector.py:170]: All the retry, with connect and refresh had been moved inside common function _query, which takes care of the retry mechanism. This exception need not be handled anywhere, and the one raised in _query will be the one to catch and validate.

File: aperturedb/Utils.py (hunk 1 above)
Comment (gsaluja9) [reply aperturedb/Utils.py:123]: oh. we do not need the = . It's just a habbit. I will remove that

**Review by drewaogle — APPROVED** [review]
lgtm

**Pull request merged**